### DISCUSSION TITLE
Conditional permissions for fields?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#211] [Q&A] [answered]
Is it possible to control read/write access to a field for an entity based on a condition? For example, how would I only allow users to add/remove roles that have a ranking <= to their current role?

**Comment by edobrb** [comment, marked as answer]
I think this is very application specific logic. The Typetta security does not provide such feature. If you are using the autogenerated operations I would suggest to override all the resolvers that does not fit your application by decorating it with your custom logic. Does it make sense?

**Reply by Atlinx** [reply]
Yea that makes sense, I ended up putting this logic inside of my security context generation code.

**Reply by minox86** [reply]
I think you can try to achieve this result in a more declarative way defining a permission connected to your logic and the defining a policy connected to a projection of the entity like described here https://twinlogix.github.io/typetta/chapters/advanced/security#permissions-on-entity-projections
